rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -208,12 +208,47 @@ public void setPendingFederation(PendingFederation federation) {
         pendingFederation = federation;
     }
 
+    @Override
+    public void clearPendingFederation() {
+        pendingFederation = null;
+    }
+
+    @Override
+    public Optional<Federation> getProposedFederation(FederationConstants federationConstants, ActivationConfig.ForBlock activations) {
+        if (!activations.isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+
+        if (proposedFederation != null || isProposedFederationSet) {
```

Comment:
Similar to [this](https://github.com/rsksmart/rskj/pull/2619/files#r1687419115), with this suggested change, I think this would change to use `&&` instead of `||`.

But let me know why we are setting `isProposedFederationSet` to `true` even when the `proposedFederation` is `null`, so I can understand better.